kms/kms.rb

```diff
@@ -0,0 +1,392 @@
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}"
+
+  # Create a KeyRing for your project
+  key_ring = kms_client.create_project_location_key_ring(resource,
+      Cloudkms::KeyRing.new, key_ring_id: key_ring_id)
+
+  puts "Created KeyRing #{key_ring_id}"
+  # [END kms_create_keyring]
+end
+
+$create_cryptokey = -> (project_id:, key_ring_id:, crypto_key:, location:) do
+  # [START kms_create_cryptokey]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of cryptoKey"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}"
+
+  # Create a CryptoKey for your project keyring
+  new_crypto_key = kms_client.create_project_location_key_ring_crypto_key(resource,
+      Cloudkms::CryptoKey.new(purpose: "ENCRYPT_DECRYPT"), crypto_key_id: crypto_key)
+
+  puts "Create CryptoKey #{crypto_key}"
+  # [END kms_create_cryptokey]
+end
+
+$encrypt = -> (project_id:, key_ring_id:, crypto_key:, location:, input_file:,
+               output_file:) do
+  # [START kms_encrypt]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # crypto_key  = "Name of the cryptoKey"
+  # location    = "The location of the new KeyRing"
+  # input_file  = "File to encrypt"
+  # output_file = "File name to use for encrypted input file"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}/" +
+             "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+  # Use the KMS API to encrypt the text
+  response = kms_client.encrypt_crypto_key(resource,
+      Cloudkms::EncryptRequest.new(plaintext: File.read(input_file)))
```

Comment:
Once I read the `decrypt` sample below (*which I love*), I love seeing the text to encrypt or decrypt called out on its own line, in its own variable.

I argued against it yesterday, but I love how this reads below:

```ruby
encrypted_text = File.read input_file

request = Cloudkms::DecryptRequest.new ciphertext: encrypted_text

response = kms_client.decrypt_crypto_key resource, request

# Write the decrypted text to a file
File.write output_file, response.plaintext
```

If you like that, let's use the same for encrypt!

_*slightly changed the syntax_